src/traffic_server/traffic_server.cc

```diff
@@ -2162,7 +2162,6 @@ main(int /* argc ATS_UNUSED */, const char **argv)
     RecProcessStart();
     initCacheControl();
     IpAllow::startup();
-    HostStatus::instance().loadHostStatusFromStats();
```

Comment:
This was key I think, it was creating the first  `HostStatus` instance, without this the ctor will not be called so no handlers registered.  You should probably need a `init()` function to register the handlers.